implementations/rust/ockam/ockam_app/src/invitations/state.rs

```diff
@@ -0,0 +1,42 @@
+use ockam_api::cloud::share::{InvitationList, ReceivedInvitation, SentInvitation};
+
+use serde::{Deserialize, Serialize};
+
+use crate::app::ModelState;
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct State<R, S> {
```

Comment:
What's the rationale behind using generics here? Do we expect to have a state with other types?

Reply:
Yeap, that's right - they could just as easily be inline types, I don't think the generics are beneficial at this point.